src/QueryClause.ts

```diff
@@ -0,0 +1,334 @@
+/**
+ * Base QueryClause implementation.
+ */
+export abstract class QueryClause {
+    abstract toJson(): any;
```

Comment:
it is same as `TriggerClause`,  follow the design of previous version. 

I think it should be public, since it is used outside the other class.